tests/integration/test_charm.py

```diff
@@ -106,11 +80,9 @@ async def test_prometheus_scrape_relation_with_prometheus_tester(
     await ops_test.model.applications[tester_app_name].remove()
     await ops_test.model.wait_for_idle(apps=[prometheus_app_name], status="active")
 
-    relation_removed_config = await get_prometheus_config(ops_test, prometheus_app_name, 0)
+    relation_removed_config, relation_removed_rules = await asyncio.gather(
+        get_prometheus_config(ops_test, prometheus_app_name, 0),
+        get_prometheus_rules(ops_test, prometheus_app_name, 0),
+    )
     assert initial_config == relation_removed_config
-
-    relation_removed_rules = await get_prometheus_rules(ops_test, prometheus_app_name, 0)
     assert initial_rules == relation_removed_rules
-
-    await ops_test.model.applications[prometheus_app_name].remove()
-    await ops_test.model.reset()
```

Comment:
I am see errors when one integration test run finishes will all tests passing then I run another immediately (log attached). I noticed in such cases the old model and application have not been completely removed and are still in the terminating stage. I wonder if it would be better to retain waits on removal of model and application.
[error-1.log](https://github.com/canonical/prometheus-operator/files/7595490/error-1.log)

Reply:
Yes, in that case, it makes sense. I'm a little bit surprised by that though as a new model is created for each test run. I'll have a look at adding the cleanup back.